services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -710,8 +717,638 @@ func TestCreateApplicationVersion(t *testing.T) {
+			}
+			commitPath := path.Join("commits", commitID[:2], commitID[2:], "applications", "app", ".empty")
+
+			transformers = append(transformers, transformer)
+			commitPaths = append(commitPaths, commitPath)
+		}
+		
+		transformers = append(transformers, &DeployApplicationVersion{
+			Environment:   envAcceptance,
+			Application:   "app",
+			Version:       30,
+			LockBehaviour: api.LockBehavior_Fail,
+		})
+
+		tcs = append(tcs, TestCase{
+			Name:                   name,
+			Transformers:           transformers,
+			ExistentCommitPaths:    commitPaths[len(commitPaths)-20:],
+			NonExistentCommitPaths: commitPaths[:len(commitPaths)-20],
+		})
+	}
+
+	{
+		name := "Create more than 20 releases of the same application but with other applications in the same commits"
+		transformers := make([]Transformer, 0)
+		commitPaths1 := make([]string, 0)
+		commitPaths2 := make([]string, 0)
+
+		firstCommitID := randomCommitID()
+		transformers = append(transformers, &CreateEnvironment{
+			Environment: "acceptance",
+			Config:      config.EnvironmentConfig{Upstream: &config.EnvironmentConfigUpstream{Environment: envAcceptance, Latest: false}},
+			
+		})
+		
+		transformers = append(transformers, &CreateApplicationVersion{
+			Application:    "app1",
+			SourceCommitId: firstCommitID,
+			Manifests: map[string]string{
+				envAcceptance: "acceptance",
+			},
+		})
+		commitPaths1 = append(commitPaths1, path.Join("commits", firstCommitID[:2], firstCommitID[2:], "applications", "app1", ".empty"))
+		transformers = append(transformers, &DeployApplicationVersion{
+			Environment:   envAcceptance,
+			Application:   "app1",
+			Version:       1,
+			LockBehaviour: api.LockBehavior_Fail,
+		})
+
+		transformers = append(transformers, &CreateApplicationVersion{
+			Application:    "app2",
+			SourceCommitId: firstCommitID,
+		})
+		commitPaths2 = append(commitPaths2, path.Join("commits", firstCommitID[:2], firstCommitID[2:], "applications", "app2", ".empty"))
+
+		for idx := 0; idx < 30; idx++ {
+			commitID := randomCommitID()
+			transformers = append(transformers, &CreateApplicationVersion{
+				Application:    "app2",
+				SourceCommitId: commitID,
+				Manifests: map[string]string{
+					envAcceptance: "acceptance",
+				},
+			})
+			commitPaths2 = append(commitPaths2, path.Join("commits", commitID[:2], commitID[2:], "applications", "app2", ".empty"))
+		}
+
+		transformers = append(transformers, &DeployApplicationVersion{
+			Environment:   envAcceptance,
+			Application:   "app2",
+			Version:       31,
+			LockBehaviour: api.LockBehavior_Fail,
+		})
+
+		tcs = append(tcs, TestCase{
+			Name:                   name,
+			Transformers:           transformers,
+			ExistentCommitPaths:    append(commitPaths1, commitPaths2[len(commitPaths2)-20:]...),
+			NonExistentCommitPaths: commitPaths2[:len(commitPaths2)-20],
+		})
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.Name, func(t *testing.T) {
+			ctx := testutil.MakeTestContext()
+			// t.Parallel()
```

Comment:
These should be able to run in parallel
```suggestion
			t.Parallel()
```